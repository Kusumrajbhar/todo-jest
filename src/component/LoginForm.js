import React, { useState } from "react";

const LoginForm = ({ submit }) => {
  const [values, setValues] = useState({
    username: "",
    password: "",
  });

  const handleChange = (e) => {
    setValues({
      ...values,
      [e.target.id]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    submit(values);
  };
  return (
    <form onSubmit={handleSubmit}>
      <div className="form-group">
        <label htmlFor="username">Username:</label>
        <input id="username" onChange={handleChange} value={values.username} />
      </div>
      <div className="from-group">
        <label htmlFor="password">Password:</label>
        <input
          className="p-3"
          id="password"
          onChange={handleChange}
          value={values.password}
        />
      </div>
      <div>
        <button type="submit">Submit</button>
      </div>
    </form>
  );
};

export default LoginForm;
