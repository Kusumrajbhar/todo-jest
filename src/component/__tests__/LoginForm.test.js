import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import LoginForm from "../LoginForm";
import renderer from "react-test-renderer";

it("should have a username and a password field, also submit button", () => {
  render(<LoginForm />);

  const usernameField = screen.getByLabelText(/username/i); // /username/i is a regex
  const passwordField = screen.getByLabelText(/password/i);
  const submitButton = screen.getByText(/submit/i);

  expect(usernameField).toBeInTheDocument(); // to check username in the document
  expect(passwordField).toBeInTheDocument();
  expect(submitButton).toBeInTheDocument();
});

it("should allow the user to submit their credential", () => {
  const submit = jest.fn();
  render(<LoginForm submit={submit} />);

  const usernameField = screen.getByLabelText(/username/i); // /username/i is a regex
  const passwordField = screen.getByLabelText(/password/i);
  const submitButton = screen.getByText(/submit/i);

  userEvent.type(usernameField, "kusum");
  userEvent.type(passwordField, "hiikusum");
  userEvent.click(submitButton);

  expect(submit).toHaveBeenCalledWith({
    username: "kusum",
    password: "hiikusum",
  });
});

test("matches snapshot", () => {
  const submit = jest.fn();
  const tree = renderer.create(<LoginForm submit={submit} />).toJSON();
  expect(tree).toMatchSnapshot();
});
