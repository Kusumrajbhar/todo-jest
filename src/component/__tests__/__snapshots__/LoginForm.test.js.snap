// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="username"
    >
      Username:
    </label>
    <input
      id="username"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="from-group"
  >
    <label
      htmlFor="password"
    >
      Password:
    </label>
    <input
      className="p-3"
      id="password"
      onChange={[Function]}
      value=""
    />
  </div>
  <div>
    <button
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
